### PR TITLE
mam2: multiple packet test

### LINKED ISSUE #898: mam2: write API test message with many packets

### PR DESCRIPTION
**Pull request by thibault-martinez**
Based on #867, will rebase when merged
Fixes #898

### CODE REVIEW

File: mam/v2/api/api.c [hunk 1]
```diff
@@ -23,20 +23,23 @@
...
-                                trit_t *address, trits_t message) {
+                                trit_t const *const address,
+                                trit_t const *const tag, trits_t message) {
   iota_transaction_t transaction;
   flex_trit_t buffer[FLEX_TRIT_SIZE_6561];
   trits_t message_part = trits_null();
 
   transaction_reset(&transaction);
-  flex_trits_from_trits(buffer, NUM_TRITS_ADDRESS, address, NUM_TRITS_ADDRESS,
+  flex_trits_from_trits(buffer, NUM_TRITS_MESSAGE, address, NUM_TRITS_ADDRESS,
```
Comment (tsvisabo) [inline mam/v2/api/api.c]: this is not a guarantee, use `trits_size`

File: mam/v2/api/api.c [hunk 2]
```diff
@@ -23,20 +23,23 @@
...
                         NUM_TRITS_ADDRESS);
   transaction_set_address(&transaction, buffer);
   transaction_set_value(&transaction, 0);
   transaction_set_obsolete_tag(&transaction,
                                transaction.data.signature_or_message);
   transaction_set_timestamp(&transaction, current_timestamp_ms() / 1000);
-  transaction_set_tag(&transaction, transaction.data.signature_or_message);
+  flex_trits_from_trits(buffer, NUM_TRITS_MESSAGE, tag, NUM_TRITS_TAG,
```
Comment (tsvisabo) [inline mam/v2/api/api.c]: ditto

File: mam/v2/mam/message.c [hunk 3]
```diff
@@ -808,7 +811,12 @@ retcode_t mam_msg_recv_packet(mam_msg_recv_context_t *ctx, trits_t *b,
   tryte_t checksum = -1;
 
   /*  absorb long trint ord; */
-  ERR_BIND_RETURN(pb3_unwrap_absorb_longtrint(&ctx->spongos, b, &ctx->ord), e);
+  {
+    trit_t ord_trits[18];
```
Comment (tsvisabo) [inline mam/v2/mam/message.c]: Can we have a this block in a function? (code duplication)

**Review by tsvisabo — CHANGES_REQUESTED** [review]

File: mam/v2/api/api.c (hunk 1 above)
Comment (thibault-martinez) [reply mam/v2/api/api.c]: Buffer is statically declared with a size of `FLEX_TRIT_SIZE_6561` which is the flex_trit equivalent of `NUM_TRITS_MESSAGE `

File: mam/v2/api/api.c (hunk 2 above)
Comment (thibault-martinez) [reply mam/v2/api/api.c]: ditto

File: mam/v2/mam/message.c (hunk 3 above)
Comment (thibault-martinez) [reply mam/v2/mam/message.c]: Will probably disappear as part of #901

**Review by tsvisabo — APPROVED** [review]

**Pull request merged**